rewrite-gradle/src/main/java/org/openrewrite/gradle/tree/ResolvedGroupArtifactVersion.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.gradle.tree;
+
+import lombok.Value;
+import lombok.With;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.maven.tree.GroupArtifact;
+import org.openrewrite.maven.tree.GroupArtifactVersion;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+@Value
+@With
+public class ResolvedGroupArtifactVersion implements Serializable {
```

Comment:
From a Gradle perspective, the dependency declaration is a requested dependency, but not necessarily the same as a resolved dependency.

This class presently has the same fields as `GroupArtifactVersion` from rewrite-maven.